pysoa/common/serializer/msgpack_serializer.py

```diff
@@ -16,32 +19,53 @@ class MsgpackSerializer(BaseSerializer):
     """
     Serializes messages to/from MessagePack.
 
-    Types supported: int, str, dict, list, tuple, bytes
-    (Note that on Python 2, str->unicode and bytes->str)
+    Types supported: int, str, dict, list, tuple, bytes, currint.Amount, datetime.date, datetime.datetime,
+    datetime.time, and decimal.Decimal (note that, on Python 2, str means unicode and bytes means str)
 
-    Note that this serializer makes no distinction between tuples
-    and lists, and will always deserialize either type as a list.
+    Note that this serializer makes no distinction between tuples and lists, and will always deserialize either type as
+    a list.
 
     It implements these custom ext types:
-        1: Datetimes with second precision.
-           Encoded as an 8-byte big-endian signed integer of the number of microseconds
-           since the epoch.
-        2: Currint amount and currency
-           Encoded as a 3-byte ASCII string of the uppercased currency code concatenated with
-           a 8-byte big-endian signed integer of the minor value.
+
+    1. Date-times with microsecond precision (must not be time zone-aware)
+
+       Encoded as an 8-byte big-endian signed integer (long long) of the number of microseconds since midnight on
+       January 1, 1970 (the Unix epoch).
+
+    2. Simple dates
+
+       Encoded big-endian as 2-byte unsigned integer year followed by 1-byte unsigned integers (chars) month and day.
+
+    3. Simple times with microsecond precision
+
+       Encoded big-endian as 1-byte unsigned integers (chars) hour, minute, and second followed by 4-byte unsigned
+       integer (long) microseconds.
+
+    4. decimal.Decimal objects
+
+       Encoded big-endian as a 2-byte unsigned integer indicating the decimal string length in bytes followed by an
+       ASCII string of that many bytes (as such, all decimals will be truncated to 65,535 total digits/bytes
+       including the optional sign and decimal point characters).
+
+    5. Currint amount and currency
+
+       Encoded as a 3-byte ASCII string of the uppercased currency code concatenated with a 8-byte big-endian signed
+       integer of the minor value.
     """
 
     mime_type = u'application/msgpack'
 
-    EXT_DATETIME = 1
+    EXT_CURRINT = 2
```

Comment:
I know PEP8 suggest sorting these by name, but I feel it _would_ make it more readable to sort this block by value

Reply:
Yeah, I went back and forth on this. Settled on alphabetization, because the `STRUCT_**` constants below these constants _only_ make sense alphabetical, and it makes sense for these related constants to have the same order as the `STRUCT_**` constants to which they relate.